date.go

```diff
@@ -31,8 +30,8 @@ type DateRule struct {
 func Date(layout string) DateRule {
 	return DateRule{
 		layout:       layout,
-		message:      "must be a valid date",
-		rangeMessage: "the data is out of range",
+		message:      messages["date"],
+		rangeMessage: messages["date_range"],
```

Comment:
This seems to be cleaner, and we can use the same pattern for other built-in rules:
```go
return DateRule{
    layout: layout,
    err: NewError("validation_date_invalid", "must be a valid date"),
    errRange: NewError("validation_date_range_invalid", "the data is out of range"),
}
```
In `Validate()`, we choose the error and parameterize it, if needed.
The `Error()` function simply updates the message in the error.

Reply:
Also, for the built-in rules, we should use a common naming convention for error code: `validation_{RuleName}_{ErrorName}`.

Reply:
So I change all validation codes to follow the same pattern.